src/runtime/metal.cpp

```diff
@@ -340,7 +340,7 @@ extern "C" {
 //   previous call (if any) has not yet been released via halide_release_metal_context.
 WEAK int halide_metal_acquire_context(void *user_context, mtl_device **device_ret,
                                       mtl_command_queue **queue_ret, bool create) {
-    halide_abort_if_false(user_context, &thread_lock != nullptr);
+    halide_debug_assert(user_context, &thread_lock != nullptr);
```

Comment:
You're probably right-- it just looks strange to potentially deref a null pointer, even if it can't be null.  If it really can't happen, I wonder if we should elide the check.